src/main/java/org/jboss/set/pull/processor/impl/evaluator/PullRequestVsStreamComponentLabelEvaluator.java

```diff
@@ -0,0 +1,69 @@
+import org.jboss.set.pull.processor.data.EvaluatorData;
+import org.jboss.set.pull.processor.data.LabelData;
+import org.jboss.set.pull.processor.data.LabelItem;
+import org.jboss.set.pull.processor.data.PullRequestData;
+
+/**
+ * Check if PR repository and branch match those defined in aphrodite.
+ *
+ * @author baranowb
+ *
+ */
+public class PullRequestVsStreamComponentLabelEvaluator extends AbstractLabelEvaluator {
+
+    @Override
+    public void eval(EvaluatorContext context, EvaluatorData data) {
+        final LabelData labelData = super.getLabelData(EvaluatorData.Attributes.LABELS_CURRENT, data);
+        final PullRequestData pullRequestData = data.getAttributeValue(EvaluatorData.Attributes.PULL_REQUEST_CURRENT);
+        final LabelData labelDataUpstream = super.getLabelData(EvaluatorData.Attributes.LABELS_UPSTREAM, data);
+        final PullRequestData pullRequestDataUpstream = data.getAttributeValue(EvaluatorData.Attributes.PULL_REQUEST_UPSTREAM);
+        scrutinize(labelData, pullRequestData);
+        scrutinize(labelDataUpstream, pullRequestDataUpstream);
+    }
+
+    protected void scrutinize(final LabelData labelData, final PullRequestData pullRequestData) {
+        try {
+            //this might happen in case of upstream either not required or
+            if (!pullRequestData.isRequired() || pullRequestData.getPullRequest() == null) {
+                return;
+            }
+
+            final URI pullRequestRepositoryURI = pullRequestData.getPullRequest().getRepository().getURL().toURI();
+            final URI componentRepositoryURI = pullRequestData.getStreamComponentDefinition().getStreamComponent().getRepositoryURL();
+            if(pullRequestRepositoryURI.equals(componentRepositoryURI)) {
+                LabelItem<?> li = new DefinedLabelItem(DefinedLabelItem.LabelContent.Component_Repository_Mismatch, LabelItem.LabelAction.REMOVE, LabelItem.LabelSeverity.OK);
+                labelData.addLabelItem(li);
+            } else {
+                LabelItem<?> li = new DefinedLabelItem(DefinedLabelItem.LabelContent.Component_Repository_Mismatch, LabelItem.LabelAction.SET, LabelItem.LabelSeverity.BAD);
+                labelData.addLabelItem(li);
+            }
+
+            if(pullRequestData.getPullRequest().getCodebase().equals(pullRequestData.getStreamComponentDefinition().getStreamComponent().getCodebase())) {
+                LabelItem<?> li = new DefinedLabelItem(DefinedLabelItem.LabelContent.Component_Branch_Mismatch, LabelItem.LabelAction.REMOVE, LabelItem.LabelSeverity.OK);
+                labelData.addLabelItem(li);
+            } else {
+                LabelItem<?> li = new DefinedLabelItem(DefinedLabelItem.LabelContent.Component_Branch_Mismatch, LabelItem.LabelAction.SET, LabelItem.LabelSeverity.BAD);
+                labelData.addLabelItem(li);
+            }
+        } catch (URISyntaxException e) {
+            super.LOG.log(Level.SEVERE, "Failed to assess repository/PR URI", e);
```

Comment:
Can this be WARNING? this is an unlikely exception URL -> URI conversion for one PR.

Reply:
since it is unlikely, I'd like it to be sever, to signify the gravity of situation